### DISCUSSION TITLE
Any way to disable window "context" menu?

### DISCUSSION THREAD

**Discussion by dexx086** [linuxmint/discussions#395] [Questions]
In Cinnamon I'm using "Super" as my "Special key to move and resize windows". Therefore, Cinnamon uses this key in combination with Middle click (so Super + Middle click) to show up the following "context" menu for the window that I'm clicking on (anywhere inside the window):
![image](https://github.com/linuxmint/discussions/assets/8817255/3d87a685-31db-4469-b20a-1ba1951bd9e7)

This is the behaviour I'd like to disable. I **don't want this menu to appear at all** because I want to bind something else for this shortcut key.
Is there any way disable this default behaviour for this shortcut? Or just simply disable the the context menu itself? Or at least have an option to detect if this is open and manually close it from a bash command (I could do it manually from my desired custom shortcut action)?
Any help would be appreciated.

**Comment by Jopp-gh** [comment]
You can modify this behaviour. Right now I'm not on my desktop but you find these options in: `System Settings> Windows`. Also look into `System Settings> Keyboard / shortcuts`.

**Reply by dexx086** [reply]
Thanks for your reply. Unfortunately I haven't seen any settings which could be related to this "Menu" feature.

On `Settings > Windows` config page there's only actions for the `Titlebar`, but not for the application's window itself.
And on the `Keyboard > Shortcuts` config page I also haven't found any shortcut that would be related. Moreover, it's not even possible to set shortcuts with a [hotkey + mouse button] combination (only keys), so I don't expect here anything with the [Super + Middle click] shortcut that I'm trying to disable.

If you have any specific config option which I might missed, I'd be glad if you share that with me.

**Comment by Jopp-gh** [comment]
No worries, `Settings > Windows > Titlebar` is the right place to deactivate this context menu. 

Just choose another option  for the action assigned when you right click on the titlebar, use `-`.

**Reply by dexx086** [reply]
Exactly, but that's **not** what I'm looking for :)
My problem is when I press [Super + Middle click] anywhere **INSIDE** a window, not on the titlebar

And the "Super" is related to the fact I specified Super as the "Special key to move and resize windows" on `Window > Behavior` tab, so with other keys, its [\<special key> + Middle click]. It's pretty much an internally handled shortcut...

**Comment by Jopp-gh** [comment]
I guess `<Super>` is the default key for moving windows, just change it to another key. 

I tried myself now and assigned another key, example:  ctrl. 
Just hold your new modifier,  click and drag -works for me. 

Btw thanks, this shortcut is handy, usually I had to pick the title bar, that was a bit annoying.

**Reply by dexx086** [reply]
But I don't want to change my "special key to move/resize" to other than Super... My only solution here if I can disable/avoid the "context" menu somehow.

I'm glad if I helped you with this shortcut, even if it's the one that annoys me :) I run into this shortcut by the fact I was using this shortcut in my previous desktop environment (KDE Plasma) for another action, and now I cannot use that action as I found no way to disable this default and internal behavior for this shortcut.

**Reply by Jopp-gh** [reply]
Ok you don't need to change super if you don't want.
But you also don't need the middle mouse click ;)

Just hold down super in the middle of any window, left mouse button click, hold and drag. Mmb+super invokes the menu. 
This for  moving windows. I never resize windows bc too much "effort" for me, :D so idk  how to help.

**Reply by dexx086** [reply]
I think you misunderstand me a little bit. I DO need the middle click, for [Super + Middle click] :) But for my custom action!
That's why it's bad that Cinnamon has this context menu bound internally for this same [Super + Middle click], so I cannot simply bind my custom action to this shortcut because when it's pressed, Cinnamon's context menu always shows up, which I want to disable.

Btw: I know how to move/resize windows with the "special key", don't get why you mentioned that.

Anyway, it doesn't look to be a simple configuration option which can easily be turned off, I guess only some system/configuration hacking could disable this context menu, but didn't find anything related to it.

**Comment by Jopp-gh** [comment]
The key combo "middle mouse button + super" is already in use and I don't think you can change these defaults. You shouldn't worry too much for "ideal" keystrokes, sometimes you've to be a bit  flexible.

My final advice (for custom actions) is to download "Easy Stroke" or "Autokey" from the software center.

**Reply by dexx086** [reply]
Thanks, I'll give a try for these shortcut tools, but I'm closing this discussion right now because I don't see any chance to be able to disable this default behavior.

Anyway, finally I managed to create a little workaround (a tiny custom shortcut utility that I wrote in c++ with handling low-level X11 events... so not working for wayland), if anybody is interested in it, just PM me. But in the future, I hope this default behavior will be configurable in case of such custom needs.
